numba/core/generators.py

```diff
@@ -60,130 +89,204 @@ def __init__(self, lower):
         self.geninfo = lower.generator_info
         self.gentype = self.get_generator_type()
         self.gendesc = GeneratorDescriptor.from_generator_fndesc(
-            lower.func_ir, self.fndesc, self.gentype, self.context.mangler)
+            lower.func_ir, self.fndesc, self.gentype, self.context.mangler
+        )
         # Helps packing non-omitted arguments into a structure
         self.arg_packer = self.context.get_data_packer(self.fndesc.argtypes)
 
         self.resume_blocks = {}
 
     @property
     def call_conv(self):
-        return self.lower.call_conv
-
-    def get_args_ptr(self, builder, genptr):
-        return cgutils.gep_inbounds(builder, genptr, 0, 1)
+        """
+        The call convention used for the generator.
 
-    def get_resume_index_ptr(self, builder, genptr):
-        return cgutils.gep_inbounds(builder, genptr, 0, 0,
-                                    name='gen.resume_index')
+        Returns
+        -------
+        CallConv
+            The call convention.
+        """
+        return self.lower.call_conv
 
-    def get_state_ptr(self, builder, genptr):
-        return cgutils.gep_inbounds(builder, genptr, 0, 2,
-                                    name='gen.state')
+    def get_pointer(self, builder, genptr, *indices):
+        """
+        Get a pointer to a specific location in the generator structure.
+
+        Parameters
+        ----------
+        builder : IRBuilder
+            The LLVM IR builder.
+        genptr : Value
+            The pointer to the generator structure.
+        indices : tuple
+            The indices to access the desired field in the structure.
+
+        Returns
+        -------
+        Value
+            The pointer to the specified field.
+        """
+        return cgutils.gep_inbounds(builder, genptr, 0, *indices)
```

Comment:
This is missing the `name` parameter entirely, I am not sure if that is a good idea.